tests/Integration/Behaviour/Features/Context/Domain/CountryFeatureContext.php

```diff
@@ -220,4 +222,35 @@ private function formatCountryDataIfNeeded(array $data)
 
         return $data;
     }
+
+    /**
+     * @When I delete country :countryReference
+     *
+     * @param string $countryReference
+     */
+    public function deleteCountry(string $countryReference): void
+    {
+        $countryId = SharedStorage::getStorage()->get($countryReference);
+
+        $this->getCommandBus()->handle(new DeleteCountryCommand((int) $countryId));
+    }
+
+    /**
+     * @Then country :countryReference should be deleted
+     *
+     * @param string $countryReference
+     */
+    public function assertCountryIsDeleted(string $countryReference): void
+    {
+        $countryId = SharedStorage::getStorage()->get($countryReference);
+
+        try {
+            $query = new GetCountryForEditing($countryId);
+            $this->getQueryBus()->handle($query);
+
+            throw new NoExceptionAlthoughExpectedException(sprintf('Country %s exists, but it was expected to be deleted', $countryReference));
+        } catch (CountryNotFoundException $e) {
+            SharedStorage::getStorage()->clear($countryReference);
```

Comment:
not sure if we need to clear the reference, but I guess it doesn't hurt :man_shrugging:

Reply:
Yeah, actually it's a good idea! Maybe we should do this more in the future this would automatically keep the tests more consistent We just need to be aware that once a reference is asserted as deleted it's the last time it can be used